tests/integration/server.py

```diff
@@ -0,0 +1,211 @@
+    # broken
+
+    # TODO (nkubala): write token to 'log_name' log, instead of stdout
+    # is this possible in non-standard (flex)???
+
+    try:
+        client = google.cloud.logging.Client()
+        gcloud_logger = client.logger(log_name)
+        gcloud_logger.log_text(token)
+    except google.cloud.exceptions.GoogleCloudError as e:
+        logging.error('Error while writing logs: {0}'.format(e))
+        raise ErrorResponse('Error while writing logs: {0}'.format(e))
+
+    logging.debug(token)
+    print(token)
+
+
+@app.route('/monitoring', methods=['POST'])
+@verify_request
+def _monitoring():
+    request_data = request.get_json()
+    name = request_data.get('name')
+    if not name:
+        raise ErrorResponse('Please provide metric name')
+    token = request_data.get('token')
+    if not token:
+        raise ErrorResponse('Please provide metric token')
+
+    try:
+        client = google.cloud.monitoring.Client()
+
+        try:
+            descriptor = client.fetch_metric_descriptor(name)
+            if descriptor is None:
+                _create_descriptor(name, client)
+        except (google.cloud.exceptions.Forbidden,
+                google.cloud.exceptions.NotFound) as ignored:  # noqa: F841
+            _create_descriptor(name, client)
+
+        _write_metric(name, client, token)
+
+    except google.cloud.exceptions.GoogleCloudError as e:
+        logging.error('Error while writing custom metric: {0}'.format(e))
+        raise ErrorResponse('Error while writing custom metric: {0}'.format(e))
+
+    return 'OK', 200
+
+
+def _write_metric(name, client, token):
+    """
+    Write a metric to Stackdriver Monitoring.
+
+    Keyword arguments:
+    name -- The name of the metric to write. Takes the form
+    'custom.googleapis.com/{metric_name}'
+    client -- the authenticated instance of a Google Cloud Client
+    token -- a random 64-bit integer token. The metric value to be written.
+
+    Once the metric is written, the test driver will retrieve all metrics
+    written with the provided name, and verify there is an entry with the
+    same value as the provided token.
+    """
+    metric = client.metric(name, {})
+    resource = client.resource('global', labels={})
+    client.write_point(metric, resource, token)
+
+
+def _create_descriptor(name, client):
+    """
+    Create a new metric descriptor.
+    This descriptor is implicitly used to write a point-value metric to
+    Stackdriver.
+    """
+    logging.info('No descriptor found with name {0}: Creating...'.format(name))
+    descriptor = client.metric_descriptor(
+        name,
+        metric_kind=google.cloud.monitoring.MetricKind.GAUGE,
+        value_type=google.cloud.monitoring.ValueType.INT64,
+        description='Test Metric')
+    descriptor.create()
+
+
+@app.route('/exception', methods=['POST'])
+@verify_request
+def _exception():
+    request_data = request.get_json()
+    token = request_data.get('token')
```

Comment:
Seems like everything needs the token, too. Put this in `verify_request`? You can pass it into the view function:

```
return f(*args, request_token=token, **kwargs)
```